src/views/ops/Template/components/Variable.vue

```diff
@@ -0,0 +1,167 @@
+      </el-table>
+      <div class="actions">
+        <el-button size="mini" type="primary" @click="onAddClick">
+          {{ $t('Add') }}
+        </el-button>
+      </div>
+      <AddVariableDialog
+        :variable="variable"
+        :variables="variables"
+        :visible.sync="addVariableDialogVisible"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import AddVariableDialog from './AddVariableDialog'
+
+export default {
+  name: 'Variable',
+  components: {
+    AddVariableDialog
+  },
+  props: {
+    value: {
+      type: [Array],
+      default: () => []
+    }
+  },
+  data() {
+    return {
+      variable: {},
+      initial: false,
+      addVariableDialogVisible: false
+    }
+  },
+  computed: {
+    variables: {
+      get() {
+        return this.value
+      },
+      set(val) {
+        this.$emit('update:value', val)
+        this.$emit('change', val)
+      }
+    }
+  },
+  watch: {
+    variables: {
+      handler(value) {
+        if (value.length > 0 || this.initial) {
+          this.$emit('input', value)
+        }
+        if (value) {
+          this.initial = true
+        }
+      },
+      immediate: true,
+      deep: true
+    }
+  },
+  methods: {
+    removeVariable(variable) {
+      this.variables = this.variables.filter((item) => {
+        if (variable.id && item.id) {
+          return item.id !== variable.id
+        } else if (variable.var_name && item.var_name) {
+          return item.var_name !== variable.var_name
+        } else {
+          return variable.name !== item.name
+        }
+      })
+    },
+    onEditClick(variable) {
+      this.variable = variable
+      setTimeout(() => {
+        this.addVariableDialogVisible = true
+      })
+    },
+    onAddClick() {
+      this.variable = null
+      setTimeout(() => {
+        this.addVariableDialogVisible = true
+      })
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+.el-data-table ::v-deep .el-table {
+  .table {
+    margin-top: 15px;
+  }
+
+  .el-table__row {
+    &.selected-row {
+      background-color: #f5f7fa;
+    }
+
+    & > td {
+      line-height: 1.5;
+      padding: 6px 0;
+      font-size: 13px;
+      border-right: none;
+
+      * {
+        vertical-align: middle;
+      }
+
+      .el-checkbox {
+        vertical-align: super;
+      }
+
+      & > div > span {
+        text-overflow: ellipsis;
+        overflow: hidden;
+        white-space: nowrap;
+      }
+    }
+  }
+
+  .el-table__header > thead > tr > th {
+    padding: 6px 0;
+    background-color: #ffffff;
+    font-size: 13px;
+    line-height: 1.5;
+    border-right: none;
+
+    .cell {
+      white-space: nowrap !important;
+      overflow: hidden;
+      text-overflow: ellipsis;
+
+      &:hover {
+        border-right: 2px solid #EBEEF5;
+      }
+    }
+  }
+}
+
+.el-data-table ::v-deep .el-table .el-table__header > thead > tr .is-sortable {
+  padding: 5px 0;
+
+  .cell {
+    padding-top: 3px !important;
+  }
+}
+</style>
```

Comment:
The given JavaScript code is a template for generating a table of variables which has some inconsistencies that can be improved upon:

* The `variables` variable should not have an empty array property (`variables`) instead it should contain an object with the key-value pairs to map variable names to variables.
  
- A better way: 
```javascript
const variables={}

```
* In line 58, we are accessing `onEditClick()` method incorrectly, it's called on click event so there must be a selector like `.add-variable-dialog`.
  
- Fixed call: 

```js
this.onEditClick(this.variable)
```

* There are few other minor typo errors around.

So these improvements will make the script more efficient and easier to read.